src/arfs/arfs_file_wrapper.ts

```diff
@@ -388,3 +395,44 @@ export class ArFSPrivateFileToDownload extends ArFSFileToDownload {
 		return writePromise.finally(this.setLastModifiedDate);
 	}
 }
+
+export abstract class ArFSFolderToDownload {
+	abstract readonly rootFolderWithPaths: ArFSPublicFileOrFolderWithPaths | ArFSPrivateFileOrFolderWithPaths;
+
+	constructor(protected readonly hierarchy: FolderHierarchy) {}
+
+	getPathRelativeToSubtree(entity: ArFSPublicFileOrFolderWithPaths | ArFSPrivateFileOrFolderWithPaths): string {
+		const rootFolderParentPath = dirname(this.rootFolderWithPaths.path);
+		const relativePath = entity.path.replace(new RegExp(`^${rootFolderParentPath}/`), '');
+		return relativePath;
+	}
+
+	ensureFolderExistence(folderPath: string, recursive = true): void {
```

Comment:
We don't want this behavior unless the user explicitly sets a flag
```suggestion
	ensureFolderExistence(folderPath: string, recursive = false): void {
```